python/tank/platform/bundle.py

```diff
@@ -595,6 +596,21 @@ def __resolve_hook_path(self, settings_name, hook_expression):
             hooks_folder = self.tank.pipeline_configuration.get_hooks_location()
             path = hook_expression.replace("{config}", hooks_folder)
             path = path.replace("/", os.path.sep)
+
+        elif hook_expression.startswith("{engine}"):
+            # look for the hook in the currently running engine
+            try:
+                engine = self.engine
+            except AttributeError:
+                raise TankError(
+                    "%s config setting %s: Could not determine the current "
+                    "engine. Unable to resolve hook path for: '%s'" %
+                    (self, settings_name, hook_expression)
+                )
+            else:
```

Comment:
yep, exactly! IMHO the else is confusing since the `except` part raises, so the code will always go into the else. FWIW I don't think we use try/else syntax anywhere in the tk code base right now.